src/renderer/ScreenController.tsx

```diff
@@ -165,20 +156,25 @@ export default class ScreenController extends Component {
       }
     }
 
-    runtime.onOpenQrUrl = processOpenQrUrl
+    runtime.onOpenQrUrl = (url: string) => {
+      processOpenQrUrl(this.context.openDialog, this.context.closeDialog, url)
```

Comment:
.. by removing `openDialog` from `window` we end up doing this now (passing `openDialog` into every method which doesn't have access to the React context / is not a component or a hook).

It doesn't necessarily need to stay like this though, some of these "helper" methods can become hooks and / or components themselves as well but I didn't wanted to touch this part as well.